br/pkg/lightning/backend/kv/sql2kv.go

```diff
@@ -361,31 +361,47 @@ func (kvcodec *tableKVEncoder) Encode(
 		j := columnPermutation[i]
 		isAutoIncCol := mysql.HasAutoIncrementFlag(col.GetFlag())
 		isPk := mysql.HasPriKeyFlag(col.GetFlag())
-		switch {
-		case j >= 0 && j < len(row):
-			value, err = table.CastValue(kvcodec.se, row[j], col.ToInfo(), false, false)
-			if err == nil {
-				err = col.HandleBadNull(&value, kvcodec.se.vars.StmtCtx)
+		value, err = func() (types.Datum, error) {
```

Comment:
Can we extract this function to the outside? For better readability and testing.